### ISSUE TITLE
[SCIDS-9] Super-Mini Batches

### ISSUE THREAD

**Issue by hatfield-c** [hatfield-c/scids#9]
We can't fit all training data in memory at once, and super batches are superior to mini-batches for gradient behavior. Split up our calls that send data to the GPU into mini batches, but conglomerate their gradients into a single step
